crates/kubelet/src/kubelet.rs

```diff
@@ -438,18 +438,17 @@ async fn on_missing_value(
             match Api::<Secret>::namespaced(client, ns).get(name).await {
                 Ok(secret) => {
                     // I am not totally clear on what the outcome should
-                    // be of a cfgmap key miss. So for now just return an
+                    // be of a secret key miss. So for now just return an
                     // empty default.
-
                     return secret
-                        .string_data
+                        .data
                         .unwrap_or_default()
-                        .get(&seckey.key)
-                        .cloned()
+                        .remove(&seckey.key)
+                        .map(|s| String::from_utf8(s.0).unwrap_or_default())
```

Comment:
Is non-ut8 data possible in this field?

Reply:
Secrets in k8s are binary blobs, so it is possible. I think we should just error pod creation if the specified secret can't be represented as an environment variable.  The error handling in this method is currently pretty lax, so this didn't attempt to fix it.

Reply:
I agree with @peterhuene here, we can improve the error handling down the road as we need it